File: 백준/Silver/1302. 베스트셀러/베스트셀러.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"sort"
)

var (
	reader = bufio.NewReader(os.Stdin)
	writer = bufio.NewWriter(os.Stdout)
)

func main() {
	defer writer.Flush()
	var n, cnt int
	var book string
	var bookList = []string{}
	fmt.Fscanf(reader, "%d\n", &n)
	books := make(map[string]int)
	for i := 0; i < n; i++ {
		fmt.Fscanf(reader, "%s\n", &book)
		if _, exists := books[book]; exists {
			books[book] += 1
		} else {
			books[book] = 1
		}
	}
	for k, v := range books {
		if v > cnt {
			book = k
			cnt = v
		} else if v == cnt {
			bookList = []string{book, k}
			sort.Strings(bookList)
			book = bookList[0]
		}
	}
	fmt.Fprintf(writer, "%s\n", book)

}

File: 백준/Silver/1302. 베스트셀러/베스트셀러.py
import sys 
input = sys.stdin.readline 
sys.setrecursionlimit(10**6)
n = int(input())
books = {}
for _ in range(n):
    s = str(input().strip('\n'))
    if s in books:
        books[s]+=1
    else:
        books[s]=1
book = ''
cnt = 0
for k, v in books.items():
    if v>cnt:
        book = k 
        cnt = v
    elif v==cnt:
        book = min(book, k) 
print(book)
